src/qibolab/transpilers/placer.py

```diff
@@ -207,30 +220,62 @@ class Backpropagation(Placer):
     Attributes:
         connectivity (networkx.Graph): chip connectivity.
         routing_algorithm (qibolab.transpilers.routing.Transpiler): routing algorithm.
-        iterations (int): number of executions of the forward and backward routing steps.
+        gates (int): Number of two qubit gates considered before finding initial layout.
+            if 'None' just one backward step will be implemented.
+            If gates is greater than the number of two qubit gates in the circuit, the circuit will be routed more than once.
     """
 
-    def __init__(self, connectivity, routing_algorithm, iterations=1):
+    def __init__(self, connectivity: nx.Graph, routing_algorithm: Router, gates=None):
         self.connectivity = connectivity
         self.routing_algorithm = routing_algorithm
-        self.iterations = iterations
+        self.gates = gates
 
     def __call__(self, circuit: Circuit):
-        """Find the initial layout of the given circuit using subgraph isomorphism.
+        """Find the initial layout of the given circuit using backpropagation placement.
 
         Args:
             circuit (qibo.models.Circuit): circuit to be transpiled.
+
+        Returns:
+            (dict): physical to logical qubit mapping.
         """
 
         initial_placer = Trivial(self.connectivity)
         initial_placement = initial_placer(circuit=circuit)
-        reversed_circuit = circuit.invert()
-        for _ in range(self.iterations):
-            final_placement = self.routing_step(initial_placement, circuit)
-            initial_placement = self.routing_step(final_placement, reversed_circuit)
-        return initial_placement
+        new_circuit = self.assemble_circuit(circuit)
+        final_placement = self.routing_step(initial_placement, new_circuit)
+        return final_placement
+
+    def assemble_circuit(self, circuit: Circuit):
+        """Assemble the initial circuit to apply backpropagation placement based on the number of two qubit gates to be considered.
+
+        Args:
+            circuit (qibo.models.Circuit): circuit to be transpiled.
 
-    def routing_step(self, layout, circuit):
+        Returns:
+            new_circuit (qibo.models.Circuit): assembled circuit to perform backpropagation placement.
+        """
+
+        if self.gates is None:
+            return circuit.invert()
+        circuit_repr = create_circuit_repr(circuit)
```

Comment:
personally, both for the variable and the function, I would prefer a more explicative name

Reply:
same for `new_circuit_repr`